kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -187,6 +188,19 @@ public Transaction build(Engine engine) {
     // tables if needed (e.g. enables column mapping)
     // Ex: We enable column mapping mode in the configuration such that our properties now include
     // Map(delta.enableIcebergCompatV2 -> true, delta.columnMapping.mode -> name)
+
+    // We must do our icebergWriterCompatV1 checks/updates FIRST since it has stricter column
+    // mapping requirements (id mode) than icebergCompatV2. It also may enable icebergCompatV2.
```

Comment:
Doesn't it have to enable icebergCompatV2? (or do you mean that may already be enabled?)

Reply:
Yeah I mean that it may additional enable it (if not already enabled). I will clarify the comment

"It will enable icebergCompatV2 if it is not already enabled" is more clear